.github/workflows/build-fdroid-app.yml

```
@@ -0,0 +1,78 @@
+name: "Prod build"
+
+on:
+  push:
+    branches:
+      - main
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.event.merge_group.head_sha }}
```

```suggestion
  group: ${{ github.workflow }}-${{ github.ref }}
```

Since this is going to run only on pushes trigger, we can use `github.ref`

Offtopic: Actually we can always use `github.ref`  in all workflows as an identifier, since it knows where it runs (pr, merge queue or branch)